### DISCUSSION TITLE
Does LDAP Integration Support Encryption

### DISCUSSION THREAD

**Discussion by btzq** [semaphoreui/semaphore#1915] [Q&A] [answered]
We are considering integrating Semaphore with our on Prem Microsoft Active Directory. 

But we are concerned about the security. Last time we checked, the AD integration with Semaphore integrates via plain text. 

Has this changed? Does it support encryption now?

**Comment by tboerger** [comment]
What do you need for that?

**Comment by fatihusta** [comment, marked as answer]
@btzq 

This may help you. [For more ldap config.](https://docs.semui.co/administration-guide/ldap
)

```
"ldap_server": "localhost:636",
"ldap_needtls": true
```

https://github.com/semaphoreui/semaphore/blob/40151c74087e6c8394e6ce4fc797dae1703cd998/api/login.go#L40
